### DISCUSSION TITLE
Using masks together with Controller component ?

### DISCUSSION THREAD

**Discussion by lucas-santosP** [react-hook-form/react-hook-form#5961] [Q&A]
Hi, i'm currently using the `Controller` component together with `Ionic` v5 inputs like this:

```js
<Controller
  name="name"
  control={control}
  render={({ field }) => (
    <IonInput type="text" onIonChange={(e) => field.onChange(e.detail.value)} />
  )}
/>;
```

Works fine but I'm having problems trying to add a mask on this. Most of the libraries that I found use their own components do to it, which in this case I don't know if would work. Since I already have a `Controller` wrapping the `IonInput`.

So do you have any indication of a library that fits well to solve this case?

**Note:** Keeping in mind that I don't intend to implement the mask manually (because of the effort to cover edge cases) and that I would like to keep using the `IonInput`.

**Comment by bluebill1049** [comment]
`Controller` should work for any controlled inputs

**Reply by lucas-santosP** [reply]
Ok but my question was more about how to use it when requires multiples wrapper components around the `input`, which one should be the last?

Anyway... I ended up deciding not to use the `IonInput` anymore because searching around I realize that it doesn't have decent support for masks when used in`React`, and looks like it was the major problem here.
